File: C#/myLeetCode/String/344. Reverse String.cs
// 344. Reverse String
// https://leetcode.com/problems/reverse-string/
public class Solution {
    public void ReverseString(char[] s) {
        int left = 0;
        int right = s.Length - 1;
        while (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;

            left++;
            right--;
        }
    }
}
public class Program {
    public static void Main() {
        Solution solution = new Solution();

        char[] input = {'h', 'e', 'l', 'l', 'o'};
        solution.ReverseString(input);

        Console.WriteLine($"Reversed string: {new string(input)}");
    }
}

File: C++/myLeetCode/String/344. Reverse String.cpp
// 344. Reverse String
// https://leetcode.com/problems/reverse-string/
#include <iostream>
#include <vector>

void reverseString(std::vector<char>& str) {
    int left = 0;
    int right = str.size() - 1;
    
    while (left < right) {
        std::swap(str[left], str[right]);
        ++left;
        --right;
    }
}

int main() {
    std::vector<char> str = {'h', 'e', 'l', 'l', 'o'};
    reverseString(str);

    for (char ch : str) {
        std::cout << ch << " ";
    }
    std::cout << std::endl;
    
    return 0;
}